test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -77,6 +77,52 @@ class CharacterTest {
     expectThrows(classOf[IllegalArgumentException], Character.toChars(Integer.MAX_VALUE))
   }
 
+  @Test def toChars_in_place(): Unit = {
+    locally {
+      val dst = new Array[Char](2)
+      assertEquals(1, Character.toChars(0x61, dst, 0))
+      assertTrue(dst sameElements Array('a', 0.toChar))
+    }
+    locally {
+      val dst = new Array[Char](2)
+      assertEquals(1, Character.toChars(0x61, dst, 1))
+      assertTrue(dst sameElements Array(0.toChar, 'a'))
+    }
+    locally {
+      val dst = new Array[Char](2)
+      assertEquals(2, Character.toChars(0x10000, dst, 0))
+      assertTrue(dst sameElements Array('\uD800', '\uDC00'))
+    }
+    locally {
+      val dst = new Array[Char](3)
+      assertEquals(2, Character.toChars(0x10001, dst, 0))
+      assertTrue(dst sameElements Array('\uD800', '\uDC01', 0.toChar))
+    }
+    locally {
+      val dst = new Array[Char](3)
+      assertEquals(2, Character.toChars(0x10401, dst, 1))
+      assertTrue(dst sameElements Array(0.toChar, '\uD801', '\uDC01'))
+    }
+    locally {
+      val dst = new Array[Char](4)
+      assertEquals(2, Character.toChars(0x10FFFF, dst, 2))
+      assertTrue(dst sameElements Array(0.toChar, 0.toChar, '\uDBFF', '\uDFFF'))
+    }
+
+    expectThrows(classOf[IllegalArgumentException], Character.toChars(Integer.MAX_VALUE, new Array(2), 0))
+
+    expectThrows(classOf[NullPointerException], Character.toChars(0x61, null, 0))
+
+    val e1 = expectThrows(classOf[IndexOutOfBoundsException], Character.toChars(0x61, new Array(2), -1))
+    assertEquals("-1", e1.getMessage)
+    val e2 = expectThrows(classOf[IndexOutOfBoundsException], Character.toChars(0x61, new Array(2), 2))
+    assertEquals("2", e2.getMessage)
+    val e3 = expectThrows(classOf[IndexOutOfBoundsException], Character.toChars(0x10FFFF, new Array(2), -1))
+    assertEquals("-1", e3.getMessage)
+    val e4 = expectThrows(classOf[IndexOutOfBoundsException], Character.toChars(0x10FFFF, new Array(2), 2))
+    assertEquals("3", e4.getMessage)
```

Comment:
Of course, you'll also have to remove the tests for NPE and IOOB (which are not valid in Scala.js because those kinds of exceptions are UB)